### PR TITLE
Robust handling of responses with invalid headers

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Thanks to @TimWolla for raising the issue.

### CODE REVIEW

**Review by TimWolla — APPROVED** [review]

File: src/Handler/HeaderProcessor.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * Returns the HTTP version, status code, reason phrase, and headers.
+     *
+     * @param string[] $hdrs
+     *
+     * @throws \RuntimeException
+     *
+     * @return array{0:string, 1:int, 2:?string, 3:array}
+     */
+    public static function parseHeaders(array $hdrs): array
```
Comment (Nyholm) [inline src/Handler/HeaderProcessor.php:21]: Let's use `$headers`

File: src/Handler/HeaderProcessor.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        if ($hdrs === []) {
+            throw new \RuntimeException('Expected a non-empty array of header data');
+        }
+
+        $parts = \explode(' ', \array_shift($hdrs), 3);
+        $ver = \explode('/', $parts[0])[1] ?? null;
```
Comment (Nyholm) [inline src/Handler/HeaderProcessor.php:28]: We can also afford to call this variable `$version`

File: src/Handler/StreamHandler.php [hunk 3]
```diff
@@ -99,11 +99,15 @@ private function createResponse(RequestInterface $request, array $options, $stre
...
-        $parts = \explode(' ', \array_shift($hdrs), 3);
-        $ver = \explode('/', $parts[0])[1];
-        $status = (int) $parts[1];
-        $reason = $parts[2] ?? null;
-        $headers = Utils::headersFromLines($hdrs);
+
+        try {
+            [$ver, $status, $reason, $headers] = HeaderProcessor::parseHeaders($hdrs);
+        } catch (\Exception $e) {
+            $msg = 'An error was encountered while creating the response';
```
Comment (Nyholm) [inline src/Handler/StreamHandler.php:106]: This variable is not really needed.

**Review by Nyholm — CHANGES_REQUESTED** [review]
I did a quick look. 

I think this is alright. 
Well done

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**